realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt

```diff
@@ -37,7 +37,9 @@ class RealmProxyMediatorGenerator(private val processingEnvironment: ProcessingE
     private val primaryKeyClasses = mutableListOf<QualifiedClassName>()
 
     init {
-        for (metadata in classesToValidate) {
+        // Sort classes to ensure deterministic output. This is relevant when e.g. using Gradle
+        // Remote Cache since the order is not guaranteed between OS and Java versions.
+        for (metadata in classesToValidate.toSortedSet(compareByDescending { it.qualifiedClassName.name })) {
```

Comment:
Any specific reason for sorting this with `compareByDescending` instead of just `compareBy`? Just makes it a bit annoying to read the expected output, but I guess we can then see it is actually affecting the output.

Reply:
Yes, I assume that the reason was that we could see it affect the output. Which it did, since it broke our unit test. But yes, sorting is otherwise arbitrary.

Properties seem to be processed in order from their class, which looks fine.